framework/test/unit/modules/interoperability/base_interoperability_module.spec.ts

```diff
@@ -681,4 +678,50 @@ must NOT have more than ${MAX_NUM_VALIDATORS} items`,
 			);
 		});
 	});
+
+	describe('finalizeGenesisState', () => {
```

Comment:
Tests added look good. Perhaps add an extra test that "if token is not native, should not throw an error".

Reply:
I see this test is added, I am resolving the comment.